### PR TITLE
cli: make --host/--{listen,advertise,http}-addr recognize port numbers

### LINKED ISSUE #23277: cli/conn+config: ux annoyance: ui shows "host:port"; cli takes --host, --port separately
While trying to decommission a dead node following the instructions in the [Remove nodes doc](https://www.cockroachlabs.com/docs/stable/remove-nodes.html#remove-a-single-node-dead), the `./cockroach node decommission 4 --wait=live --insecure --host=localhost:26257` command gave the following error even though localhost:26257 is a live node: 

```
Error: unable to connect or connection lost.
Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).
Failed to connect to the node: initial connection heartbeat failed: rpc error: code = Unavailable desc = all SubConns are in TransientFailure
Failed running "node" 
```

Whereas `./cockroach node decommission 4 --wait=live --insecure` did decommission the node.

### PR DESCRIPTION
**Pull request by knz**
cc @jseldess @Amruta-Ranade 
Fixes #23277.
Needed for #5816.

Prior to this patch, the various `cockroach` sub-commands would take
separate flags to specify an address/hostanme and to specify a port
number.

Meanwhile:

1. `--join` would recognize the syntax `host:port`.
2. the web UI, docs and other places often refer to a "server address"
   as the pair hostname:portnr.

For user convenience, it is thus important to make the interface more
straightforward/regular. This patch achieves this as follows:

- the flags
  `--listen-addr`/`--advertise-addr`/`--http-addr` (server-side) and
  `--host` (client-side) now recognize the syntax `host/addr:port`.
- the server-side `--port` flags are still recognized for backward
  compatibility but are marked as deprecated.
  The client-side `--port` is still recognized and not
  deprecated for now, but hidden from the contextual help.

As a side-effect of recognizing the port number inside the same flag,
the syntax with square brackets for IPv6 addresses now becomes
necessary when specifying also a port number. The syntax without
square brackets (and without port number) is temporarily still
recognized for backward compatibility, but is also marked as
deprecated.

Release note (cli change): the server-side command line flag
`--listen-addr`, which replaces the previous `--host` flag, is now
equipped to recognize both a hostname/address and port number. The
`--port` flag is deprecated as a result.

Release note (cli change): the server-side command line flag
`--http-addr`, which replaces the previous `--http-host` flag, is now
equipped to recognize both a hostname/address and port number. The
`--http-port` flag is deprecated as a result.

Release note (cli change): the server-side command line flag
`--advertise-addr`, which replaces the previous `--advertise-host`
flag, is now equipped to recognize both a hostname/address and
port number. The `--advertise-port` flag is deprecated as a result.

Release note (cli change): the client-side command line flag `--host`
is now equipped to recognize both a hostname/address and port
number. The client-side `--port` flag is still recognized,
but not documented any more; `--host` is now preferred.

Release note (cli change): the environment variable COCKROACH_PORT
that specifies the port number to use for client commands is now
deprecated. The port number can be placed in the COCKROACH_HOST
environment variable instead.

Release note (cli change): The syntax to specify IPv6 addresses with
the client-side command line flag `--host` is changed to use square
brackets, for example `--host=[::1]` instead of just `--host=::1`
previously. The previous syntax is still recognized for backward
compatibility but is deprecated.

Release note (cli change): the flag `--listen-port` which was
introduced in a recent change is now removed. (DOCS NOTE: remove both
this release note and the previous one that introduced --listen-port)

### CODE REVIEW

**Review by bdarnell — COMMENTED** [review]
> Reviewed 16 of 17 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cli/cliflags/flags.go, line 210 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxXQq0ywDyEUojMm-:-LJPxXQq0ywDyEUojMm0:b-h9ihad) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/cliflags/flags.go#L210)):*
> ```Go
> 	ClientHost = FlagInfo{
> 		Name:      "host",
> 		Shorthand: "s",
> ```

Why `-s`? I'd expect `-h` or `-a` (for "address"). (Oh, I see. We gave `-h` to `--help`. That's unfortunate)
___
*[pkg/cli/cliflags/flags.go, line 271 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxhWU-cSzKyJhScpd:-LJPxhWU-cSzKyJhScpe:b-6hp7lm) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/cliflags/flags.go#L271)):*
> ```Go
> 	Set = FlagInfo{
> 		Name:      "set",
> 		Shorthand: "s",
> ```

This wasn't present in 2.0, right?
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJPxvPS0BcTpRwOM1RJ:bbdsnju) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
> ```Tcl
> start_test "Check that client --port causes a deprecation warning."
> send "$argv quit --insecure --port=26257\r"
> eexpect "port has been deprecated, use --host"
> ```

It's weird that the client side still uses `--host`. Should we introduce a `--addr` flag for the client, or skip it and go all the way to using `--url` for everything on the client side?

This message should give the specific syntax to be used instead of just naming the `--host` flag. Or we should continue to support `--port` indefinitely and not deprecate it (I worry we might be too aggressive/annoying about deprecating things here. Better to support multiple redundant sets of flags than to force a lot of churn on users).

**Review by a-robinson — CHANGES_REQUESTED** [review]
> Reviewed 16 of 17 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cli/flags.go, line 176 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ2g5y6-QDXQuZrSGl:-LJQ2g5y6-QDXQuZrSGm:b-743ap8) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags.go#L176)):*
> ```Go
> 		if aerr, ok := err.(*net.AddrError); ok {
> 			if strings.HasPrefix(aerr.Err, "too many colons") {
> 				// Maybe this was an IPv6 address using the deprecated syntax
> ```

The fact that we have to handle this manually is mind-boggling. Thanks for doing it, though.
___
*[pkg/cli/flags.go, line 178 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ3Nqd4-TKnJA5dzpV:-LJQ3Nqd4-TKnJA5dzpW:b-uwxw4m) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags.go#L178)):*
> ```Go
> 				// Maybe this was an IPv6 address using the deprecated syntax
> 				// without '[...]'. Try that.
> 				maybeAddr := "[" + v + "]:" + *a.port
> ```

It's a bit scary to just assume that `a.port` is non-nil, especially in a method whose intent is to set `a.addr` and `a.port`, but I guess this struct's usage is pretty limited so it's your call whether it's worth doing anything about.
___
*[pkg/cli/flags_test.go, line 168 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4dqB-hei29rvrOw8:-LJQ4dqB-hei29rvrOw9:b-99kvfv) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/flags_test.go#L168)):*
> ```Go
> 		{[]string{"start", "--listen-addr", "192.168.0.111"}, "192.168.0.111:" + base.DefaultPort, "192.168.0.111:" + base.DefaultPort},
> 		{[]string{"start", "--listen-port", "12345"}, ":12345", ":12345"},
> 		{[]string{"start", "--listen-addr", "127.0.0.1", "--listen-port", "12345"}, "127.0.0.1:12345", "127.0.0.1:12345"},
> ```

There's still value in keeping tests around for deprecated features, I don't think we should remove all the `--listen-port` and `--advertise-port` tests  yet.
___
*[pkg/cli/flags_test.go, line 191 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4oNT1er_fQFcvzJr:-LJQ4oNT1er_fQFcvzJs:bwu0q2h) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags_test.go#L191)):*
> ```Go
> 		{[]string{"start", "--listen-addr", "::1"}, "[::1]:" + base.DefaultPort, "[::1]:" + base.DefaultPort},
> 		{[]string{"start", "--listen-addr", "2622:6221:e663:4922:fc2b:788b:fadd:7b48"},
> 			"[2622:6221:e663:4922:fc2b:788b:fadd:7b48]:" + base.DefaultPort, "[2622:6221:e663:4922:fc2b:788b:fadd:7b48]:" + base.DefaultPort},
> ```

Should we have some test cases that verify the behavior when conflicting flags are set? e.g. `--listen-addr=1.1.1.1:12345 --port=23456`
___
*[pkg/cli/cliflags/flags.go, line 271 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxhWU-cSzKyJhScpd:-LJQ5NX4EEY181CpPpXL:bf5vnj8) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/cliflags/flags.go#L271)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

This wasn't present in 2.0, right?
</blockquote></details>

Looks like it wasn't. I don't love that `-s` will have a different meaning for `cockroach start` than for client commands, though.
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJQ5hVt1FUaNA8GQG1q:b-i9nrs7) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

It's weird that the client side still uses `--host`. Should we introduce a `--addr` flag for the client, or skip it and go all the way to using `--url` for everything on the client side?

This message should give the specific syntax to be used instead of just naming the `--host` flag. Or we should continue to support `--port` indefinitely and not deprecate it (I worry we might be too aggressive/annoying about deprecating things here. Better to support multiple redundant sets of flags than to force a lot of churn on users). 
</blockquote></details>

I'd vote for keeping `--port` supported and not annoying people with a deprecation message. It's easy enough to keep working for existing users without much maintenance burden. We might want to hide it from the `--help` output, though.

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cli/flags.go, line 178 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ3Nqd4-TKnJA5dzpV:-LJTjC3C9rE4mgu0VYXS:b-k89xi4) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags.go#L178)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

It's a bit scary to just assume that `a.port` is non-nil, especially in a method whose intent is to set `a.addr` and `a.port`, but I guess this struct's usage is pretty limited so it's your call whether it's worth doing anything about.
</blockquote></details>

It's better than that: the syntax with an empty string is a valid input for the `net` package so this will succeed no matter what. Added a comment to clarify.
___
*[pkg/cli/flags_test.go, line 168 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4dqB-hei29rvrOw8:-LJTfh9E2X2cvHABGUyG:boxy8fc) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/flags_test.go#L168)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

There's still value in keeping tests around for deprecated features, I don't think we should remove all the `--listen-port` and `--advertise-port` tests  yet.
</blockquote></details>

Fair enough. I re-added them.
___
*[pkg/cli/flags_test.go, line 191 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4oNT1er_fQFcvzJr:-LJTfj_z5CcBA8zrMl95:b-f5ynef) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags_test.go#L191)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Should we have some test cases that verify the behavior when conflicting flags are set? e.g. `--listen-addr=1.1.1.1:12345 --port=23456`
</blockquote></details>

They are not conflicting -- the later flags override the earlier flags. This was already the case before. Added a test case to clarify.
___
*[pkg/cli/cliflags/flags.go, line 210 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxXQq0ywDyEUojMm-:-LJTg1akAuODH3IrCZuV:b59y3pq) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/cliflags/flags.go#L210)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Why `-s`? I'd expect `-h` or `-a` (for "address"). (Oh, I see. We gave `-h` to `--help`. That's unfortunate)
</blockquote></details>

`-h` is relatively standard for help so I'm not keen on using it.

`-s` for "server". I could also live with "-c" (connect).

`-a` would be sad to use because I can imagine a command later wanting to use `a` for "all" of something.
___
*[pkg/cli/cliflags/flags.go, line 271 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxhWU-cSzKyJhScpd:-LJTgGGGBPPvoh7hU4lS:b-2f6kr8) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/cliflags/flags.go#L271)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Looks like it wasn't. I don't love that `-s` will have a different meaning for `cockroach start` than for client commands, though.
</blockquote></details>

@a-robinson how so different? `-s` is not valid for `cockroach start`, in the same way that `-p` was not valid for it either.
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJTgZVF-xzwAHDdSy-U:bj4iy8f) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I'd vote for keeping `--port` supported and not annoying people with a deprecation message. It's easy enough to keep working for existing users without much maintenance burden. We might want to hide it from the `--help` output, though.
</blockquote></details>

@bdarnell yes the plan is to use `--url` for every client command (and deprecate/hide --host + --port) but that is a separate issue #4984 which I'd like to tackle in a separate PR. 

In the meantime I removed the deprecation warning for --port, but kept the flag hidden as Alex suggests.

**Review by bdarnell — APPROVED** [review]
> Reviewed 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/cli/cliflags/flags.go, line 210 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxXQq0ywDyEUojMm-:-LJU7sbIBrUGw4ZXnxrA:bof2swz) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/cliflags/flags.go#L210)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

`-h` is relatively standard for help so I'm not keen on using it.

`-s` for "server". I could also live with "-c" (connect).

`-a` would be sad to use because I can imagine a command later wanting to use `a` for "all" of something.
</blockquote></details>

`-h` for help is far from universal - `ls` and `grep` (to name a couple off the top of my head) have `-h` with other meanings. 

Since we want to move towards URLs for all client commands anyway, maybe this flag doesn't even need a short version. 
___
*[pkg/cli/cliflags/flags.go, line 271 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxhWU-cSzKyJhScpd:-LJU924P9G269GdGifHC:b9uvib5) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/cliflags/flags.go#L271)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

@a-robinson how so different? `-s` is not valid for `cockroach start`, in the same way that `-p` was not valid for it either.
</blockquote></details>

`-s` is `--store` for `start`
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJU7Gp5EXvwuoZxStEJ:b-dhadvs) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

@bdarnell yes the plan is to use `--url` for every client command (and deprecate/hide --host + --port) but that is a separate issue #4984 which I'd like to tackle in a separate PR. 

In the meantime I removed the deprecation warning for --port, but kept the flag hidden as Alex suggests.

</blockquote></details>

I dislike hidden flags; I think they're more confusing than just having an option that most people won't use in the help output.

**Review by a-robinson — CHANGES_REQUESTED** [review]
:lgtm: pending your last conversation or two with Ben finishing up.

> Reviewed 1 of 17 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-:-LJUoeovBh-eua_ZG8eh:bha5hfg)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/cli/flags.go, line 178 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ3Nqd4-TKnJA5dzpV:-LJUnT8_-eNtwE69ljBZ:bueq450) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags.go#L178)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

It's better than that: the syntax with an empty string is a valid input for the `net` package so this will succeed no matter what. Added a comment to clarify.
</blockquote></details>

I was thinking of code like

```
var a addrSetter
a.Set("foo")
```

In which `a.port` isn't just an empty string, but a nil pointer. But again, since the use of `addrSetter` is pretty tightly controlled to this small package, it's not a big deal.
___
*[pkg/cli/flags_test.go, line 191 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4oNT1er_fQFcvzJr:-LJUnoNe9oiMboJ8T8me:b-ktajrr) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags_test.go#L191)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

They are not conflicting -- the later flags override the earlier flags. This was already the case before. Added a test case to clarify.
</blockquote></details>

They're logically conflicting to someone who otherwise doesn't know how the implementation. Thanks for adding the test case!
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJUoPJ4DcIpPLX19y27:b-63eh7x) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

I dislike hidden flags; I think they're more confusing than just having an option that most people won't use in the help output. 
</blockquote></details>

Hm, I prefer not having flags with overlapping behavior in the help output since we then need to explain how they interact with each other or how they differ in the help output. I don't feel that strongly though, I'll defer to you guys.

**Review by knz — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/cli/flags.go, line 178 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ3Nqd4-TKnJA5dzpV:-LJV7HfX1HHrp5bUquNw:b-6h4s1o) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags.go#L178)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

I was thinking of code like

```
var a addrSetter
a.Set("foo")
```

In which `a.port` isn't just an empty string, but a nil pointer. But again, since the use of `addrSetter` is pretty tightly controlled to this small package, it's not a big deal.
</blockquote></details>

Yeah ok I'll leave it this way then.
___
*[pkg/cli/flags_test.go, line 191 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJQ4oNT1er_fQFcvzJr:-LJV7Kzy5F84mRWWOS7V:b-bavq6s) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/flags_test.go#L191)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

They're logically conflicting to someone who otherwise doesn't know how the implementation. Thanks for adding the test case!
</blockquote></details>

You're welcome.
___
*[pkg/cli/cliflags/flags.go, line 210 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxXQq0ywDyEUojMm-:-LJV7ODv8HODS4_QFSrK:b-i1zqn2) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/cliflags/flags.go#L210)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

`-h` for help is far from universal - `ls` and `grep` (to name a couple off the top of my head) have `-h` with other meanings. 

Since we want to move towards URLs for all client commands anyway, maybe this flag doesn't even need a short version. 
</blockquote></details>

Yes I also had missed that `-s` is alias for `--store`. Let's remove the short flag altogether. 
___
*[pkg/cli/cliflags/flags.go, line 271 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxhWU-cSzKyJhScpd:-LJV7TN71B3t8XmLkS71:b-i0qiu9) ([raw file](https://github.com/cockroachdb/cockroach/blob/bf76db84cb64dc90f65d8b2e129c75028127cda2/pkg/cli/cliflags/flags.go#L271)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

`-s` is `--store` for `start`
</blockquote></details>

Acknowledged. I had forgotten.
___
*[pkg/cli/interactive_tests/test_flags.tcl, line 75 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-LJPxvPS0BcTpRwOM1RI:-LJV7W8B9fZNhTkGk9BJ:b9jcy4r) ([raw file](https://github.com/cockroachdb/cockroach/blob/f180c52ffced05168f9b347208d023f648e15d1b/pkg/cli/interactive_tests/test_flags.tcl#L75)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Hm, I prefer not having flags with overlapping behavior in the help output since we then need to explain how they interact with each other or how they differ in the help output. I don't feel that strongly though, I'll defer to you guys.
</blockquote></details>

I'm with Alex and rather not list / document the flag.

**Review by bdarnell — COMMENTED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28373#-:-LJVUc6j6IxlRaUHss-x:bg7igkv)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale)

**Comment by knz** [comment]
Alex?

**Review by a-robinson — APPROVED** [review]
Weird, I thought you had my LGTM as of https://github.com/cockroachdb/cockroach/pull/28373#pullrequestreview-144955896. Github has gotten more complicated.

**Pull request merged**
